### PR TITLE
refactor: DeckPicker -  DatabaseErrorDialogType

### PR DESCRIPTION
**Pull request by david-allison**
Minor refactoring. I want to add another dialog for 'user uninstalled', and this improves things by strongly typing a parameter

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt [hunk 1]
```diff
@@ -415,29 +413,39 @@ class DatabaseErrorDialog : AsyncDialogFragment() {
...
-        const val DIALOG_LOAD_FAILED = 0
-        const val DIALOG_DB_ERROR = 1
-        const val DIALOG_ERROR_HANDLING = 2
-        const val DIALOG_REPAIR_COLLECTION = 3
-        const val DIALOG_RESTORE_BACKUP = 4
-        const val DIALOG_NEW_COLLECTION = 5
-        const val DIALOG_CONFIRM_DATABASE_CHECK = 6
-        const val DIALOG_CONFIRM_RESTORE_BACKUP = 7
-        const val DIALOG_FULL_SYNC_FROM_SERVER = 8
+    enum class DatabaseErrorDialogType(val id: Int) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:422]: Add a `@NeedsTest("assert all IDs are unique")`

**Review by BrayanDSO — APPROVED** [review]
Left only a `@NeedsTest` to add

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:422]: Fair play. Test aded

Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:422]: Just remembered that enums can be made parcelable pretty easy, which should remove the need of using an ID and `fromInt`

Leave it as a TODO if you want

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:422]: An int is required due to `Message.what` being used internally.

We could use `.ordinal`, but I've been bitten by enum reorderings before, there's the potential for a serious bug to creep in on app update (given that there's a lot of recovery screens in this enum)

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt]: I was incorrect and this will be trivial. Done

**Review by mikehardy — APPROVED** [review]
Nice

**Pull request merged**